TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -12,23 +12,38 @@ namespace TLSharp.Core.Network
 
     public class TcpTransport : IDisposable
     {
-        private readonly TcpClient tcpClient;
-        private readonly NetworkStream stream;
+        private TcpClient tcpClient;
+        private NetworkStream stream;
         private int sendCounter = 0;
+        TcpClientConnectionHandler handler;
+        string address;
+        int port;
+        IPAddress ipAddress; 
 
         public TcpTransport(string address, int port, TcpClientConnectionHandler handler = null)
+        {
+            this.handler = handler;
+            this.address = address;
+            this.port = port;
+            ipAddress = IPAddress.Parse(address);
+        }
+
+        public async Task Connect()
```

Comment:
if this method is async its name should have the `Async` suffix

Reply:
we can change, but also send and receive in the same class are async and are not named as such. i suggest to keep the same style.

Reply:
this is a convention by Microsoft, the creators of C#; if you see any current API that doesn't fit this pattern we should rename it

Reply:
ok then
guys, i think we should push this fix asap, i have been testing on a software which has been running for half a day now, perfectly handling the reconnection.

before this fix, it could last only 30 minutes.